### DISCUSSION TITLE
NubesGen JHipster microservice terraform files generation

### DISCUSSION THREAD

**Discussion by moksamedia** [microsoft/NubesGen#296] [Q&A]
Hello,

I'm a little confused about how the NubesGen GitOps generator is supposed to work. Does it only generate terraform scripts for monolithic apps? Or am I confused about how this is working? I'm trying to generate GitOps scripts for a JHipster microservice, but all the terraform scripts I see seem to only include one application service. 

Do I need to modify the generated terraform files manually for a microservice?

**Comment by jdubois** [comment]
Hi @moksamedia ! The current target is monoliths, so indeed it's a bit complicated for microservices.
As you saw, for Azure Spring Cloud it will generate the full cluster + one microservice (or monolith, you can totally deploy a monolith on Azure Spring Cloud).
Then, how do you deploy multiple microservices?

- Currently, you could clone the existing service to create other ones.
- Soon, we'll propose the option to generate other microservices and deploy them to an existing cluster.

We're a bit late on that option, as we realised we have the same issue in our upcoming AKS support, and we're trying to have a generic approach.

**Reply by moksamedia** [reply]
Thanks @jdubois. That points me in the right direction.
